src/options.h

```diff
@@ -54,11 +54,13 @@ struct ScrotOptions {
     int border;
     int silent;
     int multidisp;
-    /* If != -1, the --thumb option is in use. */
-    int thumbWorP;
-    /* If thumbH != -1, thumbWorP contains a width and thumbH contains a
-     * height. Otherwise, thumbWorP contains a percentage.
+    /* If 0, not in use.
+     * If -1, use thumbW and thumbH as the thumbnail's resolution.
+     * If between 1 inclusive and 100 inclusive, the thumbnail's resolution is
+     * opt.thumb% of the original's resolution.
      */
+    int thumb;
```

Comment:
Since goal was to stop overloading a single variable with multiple meaning, I think the following would be more useful:

```c
enum { THUMB_DISABLED, THUMB_PERCENT, THUMB_RES } thumb;
int thumbPercent;
```

With this, I don't think there needs to be any comment in the code at all, since it will be self-explanatory.